src/entities/amm/amm.ts

```diff
@@ -1569,7 +1570,12 @@ export class AMM {
   }
 
   // instant apy
-  async getInstantApy(): Promise<number> {
+  async getInstantApy(timestampInMS?: number): Promise<number> {
+    let block = (await exponentialBackoff(() => this.provider.getBlock('latest'))).number;
+    if (timestampInMS) {
```

Comment:
why don't we have ``let block = (timestampInMS) ? await getBlockAtTimestampHeuristic(this.provider, timestampInMS / 1000) : (await exponentialBackoff(() => this.provider.getBlock('latest'))).number`` ? So we avoid call to get latest block if ``timestampInMS`` is passed.